File: 1510-find-lucky-integer-in-an-array/find-lucky-integer-in-an-array.java
class Solution {
    public int findLucky(int[] arr) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int ans = -1;
        for(int i : arr){
            map.put(i,map.getOrDefault(i,0)+1);
        }
        for(Map.Entry<Integer,Integer> entry : map.entrySet()){
            if(entry.getKey()==entry.getValue()){
                ans = entry.getKey();
            }
        }
        return ans;
    }
}

File: 1510-find-lucky-integer-in-an-array/find-lucky-integer-in-an-array.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) {
        unordered_map<int,int>mpp;
        int n = arr.size();
        for(int i = 0;i<n;i++){
            mpp[arr[i]]++;
        }
        int maxi = -1;
        for(int num : arr){
            if(mpp[num] == num){
                maxi = max(maxi, num);
            }
        }
        return maxi;
    }
};